fakemails/generate-fake-mail.py

```diff
@@ -0,0 +1,216 @@
+            raise ValueError("User does not have permission to access this resource")
+        else:
+            respj = json.loads(response.text)
+            raise ValueError(f"Type: {respj[0][1]['type']}, Code: {doveadm_failure_codes(respj[0][1]['code'])}")
+    except requests.exceptions.HTTPError as errh:
+        raise Exception(errh)
+    except requests.exceptions.ConnectionError as errc:
+        raise Exception(errc)
+    except requests.exceptions.Timeout as errt:
+        raise Exception(errt)
+    except requests.exceptions.RequestException as err:
+        raise Exception(err)
+    except Exception:
+        raise
+    
+def doveadm_failure_codes(code):
+    if code == 2:
+        return "Success but mailbox changed during operation"
+    elif code == 64:
+        return "Invalid parameters"
+    elif code == 65:
+        return "Data error"
+    elif code == 67:
+        return "User does not exist"
+    elif code == 68:
+        return "User does not have session"
+    elif code == 73:
+        return "User quota is full"
+    elif code == 75:
+        return "Temporary error"
+    elif code == 77:
+        return "No permission"
+    elif code == 78:
+        return "Invalid configuration"
+    else:
+        return code
+    
+def copy_fake_tempates(account, fake_dir):
+    try:
+        if os.path.exists(f"/var/mail/{account}/Maildir/cur") == True:
+            uid, gid = get_user_infos(account)
+            subprocess.run(["/bin/bash", "./rsync.sh", account, uid, gid, fake_dir])
+        else:
+            print(f"Default folder for {account} does not exists")
+    except Exception as e:
+        print(f"Failed to retrieve account information for {account}: {e}")
+
+def exec_copy(accounts, fake_dir):
+    num_processes = 5
+
+    with ThreadPoolExecutor(max_workers=num_processes) as executor:
+        futures = [executor.submit(copy_fake_tempates, account, fake_dir) for account in accounts]
+        
+        for future in futures:
+            future.result()
+
+if __name__ == '__main__':
+    tdates = [1609459200, 1640995200, 1672531200, 1704067200, round(time.time())]
+    templates = {}
+    files_size = 0
+    accounts = []
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--accountsfile', type=str, help='List of users accounts', default=f'{os.getcwd()}/accounts')
+    parser.add_argument('--templatedir', type=str, help='Template directory', default=f'{os.getcwd()}/mail-templates')
+    parser.add_argument('--fakedir', type=str, help='Destination directory for fake emails', default=f'{os.getcwd()}/fake-mails')
+    args = parser.parse_args()
+
+    if os.path.exists(args.fakedir) == False:
+        os.mkdir(args.fakedir)
+
+    if os.path.isfile(args.accountsfile) == True:
+        with open(args.accountsfile, 'r') as f:
+            accounts = [line.strip() for line in f.readlines() if line.strip()]
+    else:
+        print(f"File {args.accountsfile} does not exists")
+        os._exit(1)
+
```

Comment:
This line is vulnerable to a shell injection attack if `account`, `uid`, `gid`, or `fake_dir` contain shell metacharacters. Consider using more secure methods to handle subprocess calls such as passing arguments as a list to `subprocess.run`.
```suggestion
subprocess.run(['/bin/bash', './rsync.sh', account, uid, gid, fake_dir], shell=False)
```